Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionRepository.kt

```diff
@@ -66,20 +67,20 @@ class SubmissionRepository @Inject constructor(
     }
 
     // to be set to true once the user has opened and viewed their test result
-    fun setViewedTestResult(type: CoronaTest.Type) {
+    suspend fun setViewedTestResult(type: CoronaTest.Type) {
         Timber.tag(TAG).v("setViewedTestResult(type=%s)", type)
-        scope.launch {
+        withContext(scope.coroutineContext) {
             val test = coronaTestRepository.coronaTests.first().singleOrNull { it.type == type }
                 ?: throw IllegalStateException("No test of type $type available")
 
             coronaTestRepository.markAsViewed(identifier = test.identifier)
         }
     }
 
-    fun refreshTest(type: CoronaTest.Type? = null) {
+    suspend fun refreshTest(type: CoronaTest.Type? = null) {
         Timber.tag(TAG).v("refreshTest(type=%s)", type)
 
-        scope.launch {
+        withContext(scope.coroutineContext) {
```

Comment:
This change is needed so that `refreshTest` rethrows the exception, right? Otherwise, we wouldn't be able to handle exceptions from the call site with try-catch - or?

Reply:
Yes, that is right , launching in appScope would not propagate crashes to try and catch. and our AppScope does not have an error handler and in this case it will crash the App

Reply:
Okay. Thanks!